plugin-scalebar/src/main/java/com/mapbox/pluginscalebar/ScaleBarOption.java

```diff
@@ -0,0 +1,307 @@
+  private boolean isMetricUnit;
+
+  public ScaleBarOption(Context context) {
+    this.context = context;
+    refreshInterval = REFRESH_INTERVAL_DEFAULT;
+    setBarHeight(R.dimen.mapbox_scale_bar_height);
+    setBorderWidth(R.dimen.mapbox_scale_bar_border_width);
+    setTextSize(R.dimen.mapbox_scale_bar_text_size);
+    setMarginTop(R.dimen.mapbox_scale_bar_margin_top);
+    setMarginLeft(R.dimen.mapbox_scale_bar_margin_left);
+    setTextBarMargin(R.dimen.mapbox_scale_bar_text_margin);
+    isMetricUnit = LocaleUnitResolver.isMetricSystem();
+    setTextColor(android.R.color.black);
+    setPrimaryColor(android.R.color.black);
+    setSecondaryColor(android.R.color.white);
+  }
+
+  /**
+   * Build a scale bar widget instance with current option settings.
+   *
+   * @return The built ScaleBarWidget instance.
+   */
+  ScaleBarWidget build() {
+    ScaleBarWidget scaleBarWidget = new ScaleBarWidget(context);
+    scaleBarWidget.setBarHeight(barHeight);
+    scaleBarWidget.setBorderWidth(borderWidth);
+    scaleBarWidget.setMarginLeft(marginLeft);
+    scaleBarWidget.setMarginTop(marginTop);
+    scaleBarWidget.setTextBarMargin(textBarMargin);
+    scaleBarWidget.setMetricUnit(isMetricUnit);
+    scaleBarWidget.setRefreshInterval(refreshInterval);
+    scaleBarWidget.setPrimaryColor(primaryColor);
+    scaleBarWidget.setSecondaryColor(secondaryColor);
+    scaleBarWidget.setTextColor(textColor);
+    scaleBarWidget.setTextSize(textSize);
+    return scaleBarWidget;
+  }
+
+  /**
+   * Set plugin's minimum refresh interval,
+   * default value is {@link #REFRESH_INTERVAL_DEFAULT}.
+   *
+   * @param refreshInterval the min refresh interval, in millisecond.
+   * @return this
+   */
+  public ScaleBarOption setRefreshInterval(int refreshInterval) {
+    this.refreshInterval = refreshInterval;
+    return this;
+  }
+
+  /**
+   * Set the text color on scale bar,
+   * default value is android.R.color.black.
+   *
+   * @param textColor the text color on scale bar.
+   * @return this.
+   */
+  public ScaleBarOption setTextColor(@ColorRes int textColor) {
+    this.textColor = context.getResources().getColor(textColor);
+    return this;
+  }
+
+  /**
+   * Set the primary color of the scale bar, will be used to draw odd index blocks,
+   * default value is android.R.color.black.
+   *
+   * @param primaryColor the primary color of the scale bar.
+   * @return this.
+   */
+  public ScaleBarOption setPrimaryColor(@ColorRes int primaryColor) {
+    this.primaryColor = context.getResources().getColor(primaryColor);
+    return this;
+  }
+
+  /**
+   * Set the secondary color of the scale bar, will be used to draw even index blocks,
+   * default value is android.R.color.white.
+   *
+   * @param secondaryColor the secondaryColor color of the scale bar.
+   * @return this.
+   */
+  public ScaleBarOption setSecondaryColor(@ColorRes int secondaryColor) {
+    this.secondaryColor = context.getResources().getColor(secondaryColor);
```

Comment:
This method is deprecated in API 23. It's better to use `ContextCompat#getColor`.